lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -142,20 +143,23 @@ def virtual_delegate_name_prefix(prefix, to)
         #   See select_from_alias for examples
 
         def virtual_delegate_arel(col, to_ref)
-          # ensure the column has sql and the association is reachable via sql
+          # Ensure the association is reachable via sql
+          #
+          # But NOT ensuring the target column has sql
+          #   to_ref.klass.arel_attribute(col) loads the target classes' schema.
+          #   This cascades and causing a race condition
+          #
           # There is currently no way to propagate sql over a virtual association
-          if to_ref.klass.arel_attribute(col) && reflect_on_association(to_ref.name)
```

Comment:
that line is asking, is this column in sql.
That forces a `to_ref.klass.load_schema!`

instead, we're hoping that the developer writing this virtual_delegate actually delegates to a valid column in the database.

I've made the mistake of delegating to a non-sql column. So I liked the instant feedback before
But now, it will blowup when you use it (because we're delaying loading the targeting class) not optimal, but good in our case.

Reply:
I'll add a few more specs around this one

Reply:
So, now, we don't check the arel_attribute(col) at all, is that ok?  Previously, you raised later on but now a nil could get through.  Is that a problem?

Reply:
`arel_attribute(:bogus)` is returning `"bogus"` - no nils in sight.